CONTRIBUTING.rst

```diff
@@ -0,0 +1,177 @@
+.. _developers-contributing:
+
+Contribute to WarpX
+===================
+
+We welcome new contributors!
+Here is how to participate to the WarpX development.
+
+Git workflow
+------------
+
+The WarpX project uses `git <https://git-scm.com>`_ for version control.
+If you are new to git, you can follow one of these tutorials:
+
+- `Learn git with bitbucket <https://www.atlassian.com/git/tutorials/learn-git-with-bitbucket-cloud>`_
+- `git - the simple guide <http://rogerdudler.github.io/git-guide/>`_
+
+Make your own fork and create a branch on it
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The basic WarpX workflow is:
+
+- Fork the main repo (or update it if you already created it).
+- Implement your changes and push them on a new branch ``<branch_name>`` on your fork.
+- Create a Pull Request from branch ``<branch_name>`` on your fork to branch `master` on the main WarpX repo.
+
+First, let us setup your local git repo. Make your own fork of the main (``upstream``) WarpX repo:
+on the `WarpX Github page <https://github.com/ECP-WarpX/WarpX>`_, press the fork button.
+Then, you can execute:
+
+.. code-block:: sh
+
+   # These 4 first lines are the same as for a standard WarpX install
+   mkdir warpx_directory
+   cd warpx_directory
+   git clone --branch master https://bitbucket.org/berkeleylab/picsar.git
+   git clone --branch development https://github.com/AMReX-Codes/amrex.git
+
+   # Clone your fork on your local computer. You can get this address on your fork's Github page.
+   git clone https://github.com/<myGithubUsername>/WarpX.git
+   cd warpx
+   # Keep track of the main WarpX repo, to remain up-to-date.
+   git remote add upstream https://github.com/ECP-WarpX/WarpX.git
+
+Now you are free to play with your fork (for additional information, you can visit the
+`Github fork help page <https://help.github.com/en/articles/fork-a-repo>`_).
+
+.. note::
+
+   You do not have to re-do the setup above every time.
+   nstead, in the future, you need to update the ``master`` branch on your fork with
+
+   .. code-block:: sh
+
+      git checkout master
+      git pull upstream master
+
+Make sure you are on WarpX ``master`` branch with
+
+.. code-block:: sh
+
+   git checkout master
+
+in the WarpX directory.
+
+Create a branch ``<branch_name>`` (the branch name should reflect the piece of code you want to add, like ``fix_spectral_solver``) with
+
+.. code-block:: sh
+
+   git checkout -b <branch_name>
+
+and do the coding you want.
+It is probably a good time to look at the `AMReX documentation <https://amrex-codes.github.io/amrex/docs_html/>`_ and at the `AMReX Doxygen reference <https://ccse.lbl.gov/pub/AMReX_Docs/index.html>`_.
```

Comment:
I would move this line to before, when you are not explaining the GitHub part in detail. Because it is really important.

Reply:
Both fine for me, I think the idea here was: get set up and ready to code, now check the code docs.